src/Microsoft.Data/DataFrame.cs

```diff
@@ -402,6 +402,127 @@ public DataFrame FillNulls(IList<object> values, bool inPlace = false)
             return ret;
         }
 
+        private void ResizeByOneAndAppend(DataFrameColumn column, object value)
+        {
+            long length = column.Length;
+            column.Resize(length + 1);
+            column[length] = value;
+        }
+
+        /// <summary> 
+        /// Appends a row inplace to the DataFrame 
+        /// </summary> 
+        /// <remarks>If a column's value doesn't match its column's data type, a conversion will be attempted</remarks> 
+        /// <remarks>If <paramref name="row"/> is null, a row of nulls is appended</remarks>
+        /// <param name="row"></param> 
+        public void Append(IEnumerable<object> row = null)
+        {
+            IEnumerator<DataFrameColumn> columnEnumerator = Columns.GetEnumerator();
+            bool columnMoveNext = columnEnumerator.MoveNext();
+            if (row != null)
+            {
+                // Go through row first to make sure there are no data type incompatibilities
+                IEnumerator<object> rowEnumerator = row.GetEnumerator();
+                bool rowMoveNext = rowEnumerator.MoveNext();
+                while (columnMoveNext && rowMoveNext)
```

Comment:
Unfortunately I have to loop over `row` twice to make sure `DataFrameColumn.Set` doesn't throw later with a type mismatch. Without this check upfront, half the columns may have appended a value before an exception is thrown.